platform/lib/routers/pages.js

```diff
@@ -56,7 +57,7 @@ function ensureFileExtension(path) {
     return path += 'index.html';
   }
 
-  if (!path.endsWith('.html')) {
+  if (!PATH_HAS_FILE_EXTENSION_PATTERN.test(path)) {
```

Comment:
I'd prefer to use `path.extname` here instead of a regex.

```
const extension = path.extname(filePath)
if (!extension) {
   ...
}
```

Reply:
Done